From: Ryan Niebur <ryanryan52@gmail.com>
Subject: by default --avoid-ldap since apache2 is the only user, and we don't
 want to add extra dependencies to other apr-utils rdepends

--- a/apu-config.in
+++ b/apu-config.in
@@ -30,7 +30,8 @@ includedir="@includedir@"
 LIBS="@APRUTIL_EXPORT_LIBS@"
 INCLUDES="@APRUTIL_INCLUDES@"
 LDFLAGS="@APRUTIL_LDFLAGS@"
-LDAP_LIBS="@LDADD_ldap@"
+ORIG_LDAP_LIBS="@LDADD_ldap@"
+LDAP_LIBS=""
 DBM_LIBS="@LDADD_dbm_db@ @LDADD_dbm_gdbm@ @LDADD_dbm_ndbm@"
 
 APRUTIL_LIBNAME="@APRUTIL_LIBNAME@"
@@ -55,7 +56,7 @@ Known values for OPTION are:
   --includedir      print location where headers are installed
   --ldflags         print linker flags
   --libs            print library information
-  --avoid-ldap      do not include ldap library information with --libs
+  --avoid-ldap      do not include ldap library information with --libs (default on OpenWrt)
   --ldap-libs       print library information to link with ldap
   --avoid-dbm       do not include DBM library information with --libs
   --dbm-libs        print additional library information to link with DBM
@@ -121,7 +122,7 @@ while test $# -gt 0; do
     flags="$flags $LDAP_LIBS $DBM_LIBS $LIBS"
     ;;
     --ldap-libs)
-    flags="$flags $LDAP_LIBS"
+    flags="$flags $ORIG_LDAP_LIBS"
     ;;
     --dbm-libs)
     flags="$flags $DBM_LIBS"
